go/cmd/dolt/commands/add.go

```diff
@@ -59,40 +61,74 @@ func (cmd AddCmd) ArgParser() *argparser.ArgParser {
 	return cli.CreateAddArgParser()
 }
 
+// Generate the query that will call the `DOLT_ADD` stored proceudre.
+// This is safe because the only inputs are flag names and validated table names.
+func generateSql(apr *argparser.ArgParseResults) string {
+	var buffer bytes.Buffer
+	var first bool
+	first = true
+	buffer.WriteString("CALL DOLT_ADD(")
+
+	write := func(s string) {
+		if !first {
+			buffer.WriteString(", ")
+		}
+		buffer.WriteString("'")
+		buffer.WriteString(s)
+		buffer.WriteString("'")
+		first = false
+	}
+
+	if apr.Contains(cli.AllFlag) {
+		write("-A")
+	}
+	if apr.Contains(cli.ForceFlag) {
+		write("-f")
+	}
+	for _, arg := range apr.Args {
+		write(arg)
+	}
+	buffer.WriteString(")")
+	return buffer.String()
+}
+
 // Exec executes the command
 func (cmd AddCmd) Exec(ctx context.Context, commandStr string, args []string, dEnv *env.DoltEnv, cliCtx cli.CliContext) int {
+	queryist, sqlCtx, closeFunc, err := cliCtx.QueryEngine(ctx)
+	if err != nil {
+
+	}
+	if closeFunc != nil {
+		defer closeFunc()
+	}
+
 	ap := cli.CreateAddArgParser()
 	helpPr, _ := cli.HelpAndUsagePrinters(cli.CommandDocsForCommandString(commandStr, addDocs, ap))
 	apr := cli.ParseArgsOrDie(ap, args, helpPr)
 
-	allFlag := apr.Contains(cli.AllFlag)
-
-	if dEnv.IsLocked() {
-		return HandleVErrAndExitCode(errhand.VerboseErrorFromError(env.ErrActiveServerLock.New(dEnv.LockFile())), helpPr)
+	for _, tableName := range apr.Args {
+		if tableName != "." && !doltdb.IsValidTableName(tableName) {
+			return HandleVErrAndExitCode(errhand.BuildDError("'%s' is not a valid table name", tableName).Build(), nil)
+		}
 	}
 
-	roots, err := dEnv.Roots(ctx)
+	_, rowIter, err := queryist.Query(sqlCtx, generateSql(apr))
 	if err != nil {
-		return HandleStageError(err)
+		cli.PrintErrln(errhand.VerboseErrorFromError(err))
+		return 1
 	}
 
-	if apr.NArg() == 0 && !allFlag {
-		cli.Println("Nothing specified, nothing added.\n Maybe you wanted to say 'dolt add .'?")
-	} else if allFlag || apr.NArg() == 1 && apr.Arg(0) == "." {
-		roots, err = actions.StageAllTables(ctx, roots, !apr.Contains(cli.ForceFlag))
-		if err != nil {
-			return HandleStageError(err)
-		}
-	} else {
-		roots, err = actions.StageTables(ctx, roots, apr.Args, !apr.Contains(cli.ForceFlag))
-		if err != nil {
-			return HandleStageError(err)
-		}
+	schema, rowIter, err := queryist.Query(sqlCtx, generateSql(apr))
+	if err != nil {
+		cli.PrintErrln(errhand.VerboseErrorFromError(err))
+		return 1
```

Comment:
Looks like you have a duplicate call for no reason. You are checking the error and then doing the same thing again, no?

Reply:
Not clear why it's showing those line.... I'm talking about the two calls to `queryist.Query(sqlCtx, generateSql(apr))`

Reply:
I'm not sure what happened here. I must have done something wrong during the rebase.

Fixed.